mockvulnerableapis/views.py

```diff
@@ -14,36 +16,39 @@
 def add_sample_data(request):
     url = request.data["url"]
     data = request.data["data"]
+    node_id = request.data.get("node_id", "x1")
     try:
-        if SampleData.objects.filter(url=url).exists():
-            SampleData.objects.update(data=json.dumps(data))
-        else:
-            SampleData.objects.create(url=url, data=json.dumps(data))
-
+        SampleData.objects.update_or_create(url=url, node_id=node_id, defaults = {"data": json.dumps(data)})
         return HttpResponse(json.dumps(request.data), status=200)
     except Exception as e:
         return HttpResponse("Error Inserting Sample Data".format(e), status=500)
 
+@async_to_sync
+async def resp_delay(api_resp, delay=0):
+    await asyncio.sleep(delay)
```

Comment:
can you confirm if this only blocks the current process, and lets other processes run

Reply:
the other requests are not blocked by this response delay